### PR TITLE
Feature/2155 write e2e tests for VA appointment request flow. References #3525 and #3526

### PR DESCRIPTION
**Pull request by vbahinwillit**
## Description
Create e2e test for VA appointment request, direct schedule, and community care workflows.

## Testing done
N/A

## Screenshots
N/A

## Acceptance criteria
- [ ] All test pass

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/e2e/community-care.e2e.spec.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    // .axeCheck('.main');
+  },
+  'Select new appointment': client => {
+    client
+      .click('#new-appointment')
+      .waitForElementVisible(continueButton, Timeouts.normal)
+      .axeCheck('.main');
+  },
+  'Choose the type of care you need': client => {
+    client.click('[value="323"]').perform(() => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/e2e/community-care.e2e.spec.js]: I don't think any of these `perform` calls are necessary, you should be able to put the code inside those callbacks inline with the rest of this test

```
client
  .click('[value="323"]')
  .click('.rjsf [type="submit"]')
  .waitForElementPresent('[value="communityCare"]', Timeouts.slow)
  .assert.containsText(
    'h1',
    'Choose where you would prefer to receive your care',
  );
```

File: src/applications/vaos/tests/e2e/community-care.e2e.spec.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  },
+  'Contact information': client => {
+    client
+      .getValue('input#root_phoneNumber', result => {
+        if (!result.value) {
+          client.setValue('input#root_phoneNumber', '5035551234');
+        }
+      })
+      .click('input#root_bestTimeToCall_morning')
+      .getValue('input#root_email', result => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/e2e/community-care.e2e.spec.js]: We have a helper called `fill(selector, value)` that will remove text before setting the value, which would let you simplify this code. I think we can always fill in the email in the test.

File: src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  },
+  'Choose a VA location for your appointment': client => {
+    client.click('[value="983"]').perform(() => {
+      client
+        .waitForElementPresent(
+          '[name = "root_vaFacility"][value = "983"]',
+          Timeouts.normal,
+        )
+        .click('[name = "root_vaFacility"][value = "983"]')
+        .pause(Timeouts.normal)
```
Comment (jbalboni) [inline src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js]: I think there's probably a way to get rid of this timeout, but it's tricky, so I'll create an issue to deal with it later.

File: src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    });
+  },
+  'How would you like to be seen?': client => {
+    client.click('input#root_visitType_0').perform(() => {
+      client
+        .click('.rjsf [type="submit"]')
+        .assert.containsText('h1', 'Contact information');
+    });
+  },
+  'Contact information': client => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js]: I think there are some sections that are common across these tests. We typically try to put those in functions in the helpers file (you can pass `client` as a parameter) and call them from each test.

**Review by jbalboni — COMMENTED** [review]
A few more comments from me, but this is looking great.

File: src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js (hunk 3 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js]: Please elaborate. I can work on it if you want.

File: src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js (hunk 4 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js]: Hey Jeff. Thanks for the review. Quick question about the comment, "there are some sections that are common across these tests.". How should this be refactored? Meaning, do you want the whole test in the helper file for the common sections?

File: src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js (hunk 3 above)
Comment (jbalboni) [reply src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js]: To get rid of it I think we'd need to wait for the continue to be enabled (it has a loading spinner for a bit after you choose the facility_, so you'd have to come up with a selector that worked for that to put in `waitForElementVisible`, and I'm not positive we can use complicated attribute selectors in Nightwatch. So there'd be some trial and error.

Comment (jbalboni) [reply src/applications/vaos/tests/e2e/direct-schedule.e2e.spec.js]: If you want to take a crack at it for 30 minutes or so, that's fine with me, but I don't want to hold up the PR for a bunch more time for something I'm not positive will work.

**Review by jbalboni — APPROVED** [review]

**Review by rianfowler — APPROVED** [review]

**Pull request merged**